external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/fetcher/KafkaRecordsFetchers.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.storm.kafka.spout.internal.fetcher;
+
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.storm.kafka.spout.KafkaPartitionReader;
+import org.apache.storm.kafka.spout.KafkaPartitionReaders;
+import org.apache.storm.kafka.spout.KafkaSpout;
+import org.apache.storm.kafka.spout.KafkaSpoutConfig;
+import org.apache.storm.task.TopologyContext;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by liurenjie on 19/12/2016.
+ */
+public final class KafkaRecordsFetchers {
+    public static <K, V> KafkaRecordsFetcher<K, V> kafkaRecordsFetcherOf(KafkaSpoutConfig kafkaSpoutConfig,
+                                                                         KafkaConsumer<K, V> consumer,
+                                                                         TopologyContext context,
+                                                                         ConsumerRebalanceListener rebalanceListener) {
+        if (kafkaSpoutConfig.getManualPartitionAssignment()) {
```

Comment:
Do we need to check these flags. i.e you are already implementing an interface KafkaRecordsFetcher and any code related to Manual or Automatic should go into respective classes. User can configure one or the other . Instead of checking if its manual or not here we should be doing with in the record fetchers .